### PR TITLE
TypeScript

### PR DESCRIPTION
**Pull request by tsa96**
Adds TypeScript compilation support!

We'll need to add the `npm run build` step to release CI, and filter out the `scripts/` directory. I'll do that in just a minute.

### Checks

-   [x] **I have thoroughly tested all of the code I have modified/added/removed to ensure something else did not break**
-   [x] I have followed [semantic commit messages](https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716) e.g. `feat: Add foo`, `chore: Update bar`, etc...
-   [x] My branch has a clear history of changes that can be easy to follow when being reviewed commit-by-commit
-   [x] My branch is functionally complete; the only changes to be done will be those potentially requested in code review
-   [x] All changes requested in review have been `fixup`ed into my original commits.
-   [x] Fully tokenized all my strings (no hardcoded English strings!!) and supplied [bulk JSON strings](https://docs.momentum-mod.org/guide/localization/#bulk-adding-terms) below

### CODE REVIEW

**Review by PeenScreeker — APPROVED** [review]
the prettier changes seem unnecessary/weird, but I don't _really_ have an opinion about it. Otherwise LGTM

**Comment by tsa96** [comment]
> the prettier changes seem unnecessary/weird, but I don't _really_ have an opinion about it. Otherwise LGTM

Sorry, what Prettier changes?

**Comment by tsa96** [comment]
> > > the prettier changes seem unnecessary/weird, but I don't _really_ have an opinion about it. Otherwise LGTM
> > 
> > 
> > Sorry, what Prettier changes?
> 
> [12993c9](https://github.com/momentum-mod/panorama/pull/103/commits/12993c9959deba485d4aa5c886b5f7ed51cb0c1e)
> 
> Not changes you made to prettier, but rather the autoformatting prettier did.

Oh haha, yeah I was a bit put out by that.

File: scripts/util/dont-show-again.ts [hunk 1]
```diff
@@ -1,31 +1,28 @@
...
-	 * @param {String} key
-	 * @returns {Boolean} Whether the DoSA was deleted
+	 * Remove a stored DoSA
 	 */
-	static removeDosa(key) {
-		const dosas = this.#getPSObject();
-
+	static removeDosa(key: string): boolean {
+		const dosas = this.getPSObject();
+    if (!dosas) return false;
```
Comment (Gocnak) [inline scripts/util/dont-show-again.ts:21]: Spaces? Does prettier enforce tabs vs spaces at all?

Comment (tsa96) [reply scripts/util/dont-show-again.ts]: Was baffled as to why Prettier wasn't catching this in CI then realized I hadn't configured it to test `.ts` files in `package.json`, fixed now!

**Review by Gocnak — APPROVED** [review]

**Pull request merged**